tests/test_cli_compile.py

```diff
@@ -3001,6 +3001,40 @@ def test_cli_compile_strip_extras(runner, make_package, make_sdist, tmpdir):
     assert "[more]" not in out.stderr
 
 
+def test_cli_compile_all_extras_with_multiple_packages(
+    runner, make_package, make_sdist, tmpdir
+):
+    """
+    Assures that ``--all-extras`` works when multiple sources are specified.
+    """
+    test_package_1 = make_package(
+        "test_package_1",
+        version="0.1",
+        extras_require={"more": []},
+    )
+    test_package_2 = make_package(
+        "test_package_2",
+        version="0.1",
+        extras_require={"more": []},
+    )
+
+    out = runner.invoke(
+        cli,
+        [
+            "--all-extras",
+            "--output-file",
+            "requirements.txt",
+            str(test_package_1 / "setup.py"),
+            str(test_package_2 / "setup.py"),
+        ],
+    )
+
+    assert out.exit_code == 0, out
+    assert "--all-extras" in out.stderr
+    assert "test_package_1/0.1/setup.py" in out.stderr
+    assert "test_package_2/0.1/setup.py" in out.stderr
```

Comment:
Obviously, Windows isn't happy with POSIX file path syntax.

```suggestion
    assert str(test_package_1 / "setup.py") in out.stderr
    assert str(test_package_2 / "setup.py") in out.stderr
```

Reply:
Alternatively, just replace the path separators in the strings:
```suggestion
    assert f"test_package_1{os.path.sep}0.1{os.path.sep}setup.py" in out.stderr
    assert f"test_package_2{os.path.sep}0.1{os.path.sep}setup.py" in out.stderr
```

But the pathlib approach is fine. Though, I wouldn't copy-paste the path construction as the same thing is present in code above — it's reasonable to put that into vars and reuse across the test function.

Reply:
I went with `os.path.sep` since the other alternative uses the absolute path and would require calling `relative_to` or similar, which I think obscures the readability of what is actually tested.

Pushed in fixup now.